index.js

```diff
@@ -1,95 +1,122 @@
 // ***** Question 1 *****
+var counter = 1;
+counter = 2;
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
-
-// console.log(counter) 
-// // => 2
-// console.log("%c----------", "color: red") 
+console.log("%cQuestion 1", "color: red") 
 
+console.log(counter) 
+// => 2
+console.log("%c----------", "color: red") 
 
 // ***** Question 2 *****
 
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
-// // => "Ian"
-// name = "Not Ian" 
-// // => TypeError
-// console.log("%c----------", "color: red") 
+const name = 'SunJet';
 
+console.log("%cQuestion 2", "color: red") 
 
+console.log(name)
+// => "Ian"
+//name = "Not Ian" 
+// => TypeError
+console.log("%c----------", "color: red") 
 
 // ***** Question 3 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
-
-// console.log(drinkWater(12))
-// // => "Man I sure am thirsty"
-// // => "Ahh that hits the spot"
-// // => 11
-// console.log("%c----------", "color: red") 
-
+function drinkWater(currentThirstLevel) {
+    console.log(`"Man I sure am thirsty"`)
+    currentThirstLevel -= 1
+    console.log(`"Ahh that hits the spot"`)
+    return currentThirstLevel
+}
+    
+console.log("%cQuestion 3", "color: red") 
 
+console.log(drinkWater(12))
+// => "Man I sure am thirsty"
+// => "Ahh that hits the spot"
+// => 11
+console.log("%c----------", "color: red") 
 
 // ***** Question 4 *****
+function sameSameButDifferent(num,maybeNum){
+  if (num === maybeNum) {
+    return '"same same"';
+  }
+  else if (num === parseInt(maybeNum)) {
+    return '"same same (but different)"';
+  } 
+  else {
+    return '"different"';
+  }
+}
```

Comment:
Good use of strict (`===`) equality and `parseInt`. The return values don't need to be inside two sets of quotes - just one or the other will work fine! Also, just a small nitpick on the formatting - `else` in Javascript is typically written on the same line as the end of the `if` block, like this:

```js
  if (num === maybeNum) {
    return "same same"
  } else if (num == maybeNum) {
    return "same same (but different)"
  } else {
    return "different"
  }
```